### PR TITLE
fix(data_wizard): Make audit and risk assessment perimeter optional

### PR DESCRIPTION
**Pull request by monsieurswag**
Currently CLI users are forced to specify a `Perimeter` when using `import-risk-assessment` or `import-compliance-assessment` commands.
This is wrong as both the `RiskAssessment.perimeter` and `ComplianceAssessment.perimeter` fields are optional (they are `ForeignKey` with `null=True`).

Now users can either specify a perimeter, or only specify the folder if they want no perimeter.

## Summary by CodeRabbit

* **New Features**
  * Import commands for ComplianceAssessment and RiskAssessment no longer require a perimeter identifier; they can run using a folder identifier when perimeter is unavailable.

* **Bug Fixes**
  * Import flows now handle missing perimeter input gracefully by resolving folder context when possible, recording failures with clear errors, and exiting cleanly instead of raising exceptions.

### CODE REVIEW

File: cli/clica.py [hunk 1]
```diff
@@ -325,7 +325,7 @@ def upload_data_wizard_file(
         "model_type": "ComplianceAssessment",
         "help": "Import compliance assessments using the Data Wizard backend.",
         "requires_folder": False,
-        "requires_perimeter": True,
+        "requires_perimeter": False,
```
Comment (ab-smith) [inline cli/clica.py:331]: almost, now we're in a situation where domain (folder) and perimeter are optional

**Review by JiwaniZakir — COMMENTED** [review]
The use of `AssertionError` in both `_process_compliance_assessment` and `_process_risk_assessment` is inappropriate for input validation — `AssertionError` is intended for internal invariant checks and can be silently disabled when Python runs with the `-O` optimization flag. A `ValueError` (or a domain-specific exception) would be more semantically correct and safer here.

More critically, in `cli/clica.py`, both model type configs still have `"requires_folder": False` even after setting `"requires_perimeter": False`. This means a user can now invoke the CLI without providing either a perimeter or a folder, and the error won't surface until the request hits the backend and raises that `AssertionError`. The CLI should enforce the "at least one of perimeter or folder is required" constraint at the argument-parsing level, or at minimum `requires_folder` should be conditionally tied to whether a perimeter is provided.

It's also worth adding a test scenario where `perimeter_id=None` and `folder_id=None` are both passed to each function, to verify the error path behaves as expected rather than producing an unhandled `ObjectDoesNotExist` or silent failure.

File: backend/data_wizard/views.py [hunk 2]
```diff
@@ -2529,8 +2529,16 @@ def _process_compliance_assessment(
...
+            perimeter = None
+            if perimeter_id is not None:
+                perimeter = Perimeter.objects.get(id=perimeter_id)
+
+            if perimeter is not None:
+                folder_id = perimeter.folder.id
+            elif folder_id is None:
+                raise AssertionError(
+                    "A folder must be specified when there's no perimeter!"
+                )
```
[the remark this answers is not in the document]
Comment (nas-tabchiche) [reply backend/data_wizard/views.py:2584]: this should be addressed @monsieurswag

**Review by nas-tabchiche — CHANGES_REQUESTED** [review]

**Comment by monsieurswag** [comment]
> The use of `AssertionError` in both `_process_compliance_assessment` and `_process_risk_assessment` is inappropriate for input validation — `AssertionError` is intended for internal invariant checks and can be silently disabled when Python runs with the `-O` optimization flag. A `ValueError` (or a domain-specific exception) would be more semantically correct and safer here.
> 
> More critically, in `cli/clica.py`, both model type configs still have `"requires_folder": False` even after setting `"requires_perimeter": False`. This means a user can now invoke the CLI without providing either a perimeter or a folder, and the error won't surface until the request hits the backend and raises that `AssertionError`. The CLI should enforce the "at least one of perimeter or folder is required" constraint at the argument-parsing level, or at minimum `requires_folder` should be conditionally tied to whether a perimeter is provided.
> 
> It's also worth adding a test scenario where `perimeter_id=None` and `folder_id=None` are both passed to each function, to verify the error path behaves as expected rather than producing an unhandled `ObjectDoesNotExist` or silent failure.

The `-O` optimization flag only affects the `assert` keyword itself, not the `AssertionError` exception.
As you can see in the following output the CPython internal compiler doesn't generate any python bytecode for the `assert {...}` line in optimized mode (it completely ignores it).
But it does compiles direct `raise AssertionError(...)` normally in both optimized and normal modes.

So raising an `AssertionError` directly is not an issue.

`py -O -c 'assert False, "Some Error"'`

(No output, so no Exception raised)

 `py -O -c 'raise AssertionError("Some Error")' `

```py
Traceback (most recent call last):
  File "<string>", line 1, in <module>
    raise AssertionError("Some Error")
AssertionError: Some Error
```

`py -O`

```py
Python 3.14.2 (main, Jan 24 2026, 16:32:42) [GCC 13.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dis
>>> def f(): raise AssertionError("WOW")
... 
>>> dis.dis(f)
  1           RESUME                   0
              LOAD_GLOBAL              1 (AssertionError + NULL)
              LOAD_CONST               0 ('WOW')
              CALL                     1
              RAISE_VARARGS            1
>>> def f(): assert False, "WOW"
... 
>>> dis.dis(f)
  1           RESUME                   0
              LOAD_CONST               0 (None)
              RETURN_VALUE
```

`py`
```py
Python 3.14.2 (main, Jan 24 2026, 16:32:42) [GCC 13.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dis
>>> def f(): raise AssertionError("WOW")
... 
>>> dis.dis(f)
  1           RESUME                   0
              LOAD_GLOBAL              1 (AssertionError + NULL)
              LOAD_CONST               0 ('WOW')
              CALL                     1
              RAISE_VARARGS            1
>>> def f(): assert False, "WOW"
... 
>>> dis.dis(f)
  1           RESUME                   0
              LOAD_COMMON_CONSTANT     0 (AssertionError)
              LOAD_CONST               1 ('WOW')
              CALL                     0
              RAISE_VARARGS            1
```

---

For the doubly optional folder and perimeter i was aware of this issue, i prefered transfering the validation logic to the backend instead of breaking the "clean" DRY pattern we use to generate command in the CLI.

Note that the backend error message is clear enough for user experience to not be affected by this.

I don't remember the exact detail of the reasoning that led me to took this decision at that time, but i remember it was a well thought decision (not something i overlooked).

---

Adding a test with the sole purpose of ensuring there's no regression on a single invalid condition concerning 2 specific data_wizard objects in the backend is overkill.

If we someday add API tests for the data_wizard they shall have a way bigger coverage than just checking a single validation condition like this one.

**Review by nas-tabchiche — APPROVED** [review]
LGTM
